### PR TITLE
ci: Release to Nuget.org

### PR DESCRIPTION
**Pull request by Anush008**
This PR adds a release workflow for publishing and test workflow for PRs.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -1,12 +1,12 @@
-name: Build
+name: Test
```
Comment (IvanPleshkov) [inline .github/workflows/test.yml:1]: I guess we may revert name back, there is no reason to rename this job

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      shell: bash
+
+    - name: Pack
+      run: dotnet pack -c Release -o nuget --no-build --nologo
+      shell: bash
+
+    - name: Publish NuGet package
+      env:
+        NUGET_API_KEY: ${{ secrets.NUGET_API_KEY }}
+      run: dotnet nuget push nuget/*.nupkg --api-key $NUGET_API_KEY --source https://api.nuget.org/v3/index.json
```
Comment (IvanPleshkov) [inline .github/workflows/release.yml:34]: I guess API key is already presented

**Review by IvanPleshkov — APPROVED** [review]
I don't have enough expertise to say that publishing is correct. Not sure if this build will publish QDrant client for all supported dotnet versions. I guess just publish as .NET 6.0 is enough because it's .Net 2.0 Standart - but it's just a guess game. I place approve because this changes blocks client release

File: .github/workflows/test.yml (hunk 1 above)
Comment (Anush008) [reply .github/workflows/test.yml]: https://github.com/qdrant/qdrant-dotnet/pull/48/commits/65c68a95575fc851ef7dac95874573ce65c93f0a

File: .github/workflows/release.yml (hunk 2 above)
Comment (Anush008) [reply .github/workflows/release.yml:34]: Should we try to do a release? Assuming the secret is set?

**Pull request merged**